homeassistant/components/zha/sensor.py

```diff
@@ -7,20 +7,105 @@
 import logging
 
 from homeassistant.components.sensor import DOMAIN
-from homeassistant.const import TEMP_CELSIUS
+from homeassistant.const import STATE_UNKNOWN
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
-from homeassistant.util.temperature import convert as convert_temperature
-from .core import helpers
 from .core.const import (
-    DATA_ZHA, DATA_ZHA_DISPATCHERS, REPORT_CONFIG_MAX_INT,
-    REPORT_CONFIG_MIN_INT, REPORT_CONFIG_RPT_CHANGE, ZHA_DISCOVERY_NEW)
+    DATA_ZHA, DATA_ZHA_DISPATCHERS, ZHA_DISCOVERY_NEW, HUMIDITY, TEMPERATURE,
+    ILLUMINANCE, PRESSURE, METERING, ELECTRICAL_MEASUREMENT,
+    POWER_CONFIGURATION, GENERIC, SENSOR_TYPE, LISTENER_ATTRIBUTE,
+    LISTENER_ACTIVE_POWER, SIGNAL_ATTR_UPDATED, SIGNAL_STATE_ATTR)
 from .entity import ZhaEntity
 
 _LOGGER = logging.getLogger(__name__)
 
 DEPENDENCIES = ['zha']
 
 
+# unit getters
+def get_temperature_unit(sensor):
```

Comment:
I would think that we should return the unit of the device, not the unit of home assistant.

Reply:
This is how it works today in ZHA in HA. We can explore that in the future.